### PR TITLE
feature/add_remove_workers

### PR DESCRIPTION
**Pull request by quasiben**
Tied to PR https://github.com/dask/knit/pull/61

### CODE REVIEW

**Comment by quasiben** [comment]
@mrocklin when you have some time can you look this over ?

**Comment by mrocklin** [comment]
Ah, yes sorry.  I did give this a shot and everything work smoothly for me.  Thanks as always for maintaining this.  It's a great help.  I think I forgot to mention that things were working well here only because I became too distracted by the results that I was seeing on the experiment I was running :)

**Comment by quasiben** [comment]
Is it possible you think you are commenting on https://github.com/dask/dask-ec2/pull/76 ?

**Comment by mrocklin** [comment]
ah yes, indeed.  My mistake

On Mon, Mar 27, 2017 at 10:03 PM, Benjamin Zaitlen <notifications@github.com

File: dask_yarn/core.py [hunk 1]
```diff
@@ -59,12 +63,68 @@ def start(self, n_workers, cpus=1, memory=4000):
...
+        pass
+
     def stop(self):
         try:
-            self.knit.kill(self.app_id)
+            self.knit.kill()
         except AttributeError:
             pass
 
+    def add_workers(self, n_workers=1, cpus=1, memory=512):
```
Comment (mrocklin) [inline dask_yarn/core.py]: I recommend a couple gigabytes of ram by default if that's reasonable

File: dask_yarn/tests/test_core.py [hunk 2]
```diff
@@ -92,7 +93,41 @@ def test_yarn_cluster_stop(loop):
...
+    num_containers = status['app']['runningContainers']
+    assert num_containers == 2  # 1 container for the worker and 1 for the RM
+
+    cluster.add_workers(n_workers=1, cpus=1, memory=256)
+
+    while num_containers != 3:
+        status = cluster.knit.status()
+        num_containers = status['app']['runningContainers']
+
+    # wait a tad to let workers connect to scheduler
```
Comment (mrocklin) [inline dask_yarn/tests/test_core.py:106]: Perhaps sleep a bit in a while loop?

```python
start = time()
while len(client.scheduler_info()['workers']) < 2:
    sleep(0.1)
    assert time() < start + 10
```

Comment (mrocklin) [reply dask_yarn/tests/test_core.py:106]: Travis.ci is sometimes slow

**Review by mrocklin — COMMENTED** [review]
These seem fine to me

File: dask_yarn/tests/test_core.py (hunk 2 above)
Comment (quasiben) [reply dask_yarn/tests/test_core.py:106]: Thanks!

**Pull request merged**